esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1335,6 +1394,40 @@ bool CWsESDLConfigEx::onGetESDLDefinition(IEspContext &context, IEspGetESDLDefin
     }
 
     resp.setXMLDefinition(definition.str());
+    if (ver >= 1.2)
+    {
+        if (req.getReportMethodsAvailable())
+        {
+            if (definition.length() > 0)
+            {
+                try
+                {
+                    Owned<IPropertyTree> definitionTree = createPTreeFromXMLString(definition.str(), ipt_caseInsensitive);
+                    Owned<IPropertyTreeIterator> iter = definitionTree->getElements("EsdlMethod");
+                    IArrayOf<IEspMethodConfig> list;
+                    ForEach(*iter)
+                    {
+                        Owned<IEspMethodConfig> methodconfig = createMethodConfig("","");
+                        IPropertyTree &item = iter->query();
+                        methodconfig->setName(item.queryProp("@name"));
+                        list.append(*methodconfig.getClear());
+                    }
+                    resp.setMethods(list);
+                }
+                catch (...)
+                {
+                    message.append("\n Encountered error while parsing fetching available methods");
```

Comment:
yes... this portion is an "add-on" so the request could be successful even though not all the information was fetch-able...